src/colorizer/utils/data_utils.ts

```diff
@@ -327,3 +328,41 @@ export function getGlobalIdFromSegId(
 
   return rawGlobalId - 1; // -1 to convert to zero-based index
 }
+
+export function getLabelTypeFromParsedCsv(
+  headers: string[],
+  data: Record<string, string | undefined>[]
+): Map<string, LabelType> {
+  const labelTypeMap = new Map<string, LabelType>();
+  for (const header of headers) {
+    let hasIntegerValues = true;
+    let hasBooleanValues = true;
+    for (const row of data) {
+      const value = row[header]?.trim();
+      const valueAsInt = parseInt(value ?? "", 10);
+      if (value === undefined || value === "") {
+        continue;
+      } else if (value.toLowerCase() === BOOLEAN_VALUE_TRUE || value.toLowerCase() === BOOLEAN_VALUE_FALSE) {
+        hasIntegerValues = false;
+      } else if (valueAsInt.toString(10) === value && Number.isInteger(valueAsInt)) {
+        hasBooleanValues = false;
+      } else {
+        hasBooleanValues = false;
+        hasIntegerValues = false;
+        break;
+      }
+      if (!hasIntegerValues && !hasBooleanValues) {
+        break;
+      }
```

Comment:
are these two checks doing the same work? If L354 check is necessary, would moving this check up a few lines as an early exit cleaner?

Reply:
Mm, we do still need it. The `else` in L350 triggers if a value is not a boolean and not an integer (e.g. a string). The check on L354 will trigger if a column has mixed booleans AND integers at the same time.